### ISSUE TITLE
Auto-completion will select all text before the first argument when toggled for a function

### ISSUE THREAD

**Issue by greyishsong** [neoclide/coc.nvim#4599]
## Result from CocInfo

```
## versions

vim version: VIM - Vi IMproved 8.2 8024919
node version: v16.15.1
coc.nvim version: 0.0.82-b7375d5f 2023-01-30 05:09:03 +0800
coc.nvim directory: /home/greyishsong/.vim/plugged/coc.nvim
term: dumb
platform: linux

## Log of coc.nvim

2023-04-09T21:10:46.294 INFO (pid:35307) [plugin] - coc.nvim initialized with node: v16.15.1 after 184
2023-04-09T21:10:46.298 WARN (pid:35307) [workspace] - workspace.createStatusBarItem is deprecated, please use window.createStatusBarItem instead. 
    at Object.t.initStatusBar (/home/greyishsong/.config/coc/extensions/node_modules/coc-spell-checker/out/index.js:1:156009)
    at Object.t.activate (/home/greyishsong/.config/coc/extensions/node_modules/coc-spell-checker/out/index.js:1:23162)
2023-04-09T21:10:46.308 INFO (pid:35307) [language-client-index] - Language server "cSpell" started with 35321
2023-04-09T21:10:46.323 INFO (pid:35307) [services] - LanguageClient clangd state change: stopped => starting
2023-04-09T21:10:46.326 INFO (pid:35307) [language-client-index] - Language server "clangd" started with 35328
2023-04-09T21:10:46.359 INFO (pid:35307) [services] - LanguageClient clangd state change: starting => running
2023-04-09T21:10:46.363 INFO (pid:35307) [services] - service clangd started
2023-04-09T21:10:49.617 INFO (pid:35307) [attach] - Request action: listNames []
2023-04-09T21:10:49.974 INFO (pid:35307) [attach] - receive notification: openList [ 'extensions' ]
2023-04-09T21:10:49.991 INFO (pid:35307) [attach] - receive notification: doAutocmd [ 2, 2 ]
2023-04-09T21:10:49.992 INFO (pid:35307) [attach] - receive notification: doAutocmd [ 1 ]
2023-04-09T21:11:00.083 INFO (pid:35307) [attach] - receive notification: showSignatureHelp []
2023-04-09T21:11:04.631 INFO (pid:35307) [attach] - receive notification: showInfo []
```

## Describe the bug

When I try to complete any C/C++ function with argument(s), coc.nvim autometically selects all text before (and above) the first argument together with the first argument itself like this:

![Screenshot_20230409_205620](https://user-images.githubusercontent.com/49446254/230774801-54bceff4-73b6-4711-b153-cd62ba10bbd5.png)
![Screenshot_20230409_205644](https://user-images.githubusercontent.com/49446254/230774820-9f612f58-1bbe-44ca-8cdf-99fb03e7399a.png)

If I type something now, all selected text are replaced:

![Screenshot_20230409_205701](https://user-images.githubusercontent.com/49446254/230774998-e1a79d63-b499-401d-b031-bd4d2e1705bd.png)

Theoretically, only the first argument should be selected and replaced.

## Reproduce the bug

- Install coc.nvim via vim-plug
- Install coc-clangd (with clangd-14 on Ubuntu installed by apt package manager)
- Replace the user's `~/.vimrc` with：
  ```vim
  call plug#begin('~/.vim/plugged')
  Plug 'neoclide/coc.nvim', {'branch': 'release'}
  call plug#end()

  inoremap <silent><expr> <CR> coc#pum#visible() ? coc#pum#confirm()
                                \: "\<C-g>u\<CR>\<c-r>=coc#on_enter()\<CR>"

  function! CheckBackspace() abort
    let col = col('.') - 1
    return !col || getline('.')[col - 1]  =~# '\s'
  endfunction

  augroup mygroup
    autocmd!
    autocmd FileType typescript,json setl formatexpr=CocAction('formatSelected')
    autocmd User CocJumpPlaceholder call CocActionAsync('showSignatureHelp')
  augroup end
  ```
- Open any C/C++ source file and try to toggle auto-completion for a function with at least one argument.

## Additions

If I remove these from the vimrc file, coc.nvim works correctly:

```vim
augroup mygroup
  autocmd!
  autocmd FileType typescript,json setl formatexpr=CocAction('formatSelected')
  autocmd User CocJumpPlaceholder call CocActionAsync('showSignatureHelp')
augroup end

```

I have other coc extensions installed, but only coc-clangd enabled for all tests mentioned above:

![Screenshot_20230409_205551](https://user-images.githubusercontent.com/49446254/230775453-5537a923-b085-4f87-b05f-ed8ca57876e3.png)

My coc-settings.json contains only `"coc.preferences.extensionUpdateCheck": "daily"`.

**Comment by fannheyward** [comment]
Can't reproduce with vim 9.0, maybe something didn't work perfectly with vim 8.2.

Try `"signature.target": "echo"` in your coc-settings.json, this will display function signature in `:messages`, present in the floating window.

Or upgrade your vim and try again.

**Comment by greyishsong** [comment]
> Can't reproduce with vim 9.0, maybe something didn't work perfectly with vim 8.2.
> 
> Try `"signature.target": "echo"` in your coc-settings.json, this will display function signature in `:messages`, present in the floating window.
> 
> Or upgrade your vim and try again.

Thanks for your effort and advice!

I tried to disable other coc extensions globally, and I found that disabling coc-explorer globally reduces some error (but not all). Now I am trying to reproduce the hehavior on a clean virtual machine (with vim 8.2). If I success, I will try to provide more details about it.

For my physical machine, today I update my vim to 9.0.1443 and everything works correctly. Vim 9.0 performs mach better than 8.2, thanks again for advising me to upgrade.
